src/NServiceBus.Metrics/MetricsOptions.cs

```diff
@@ -65,14 +65,16 @@ public void EnableCustomReport(Func<string, Task> func, TimeSpan interval)
         /// </summary>
         /// <param name="serviceControlMetricsAddress">The transport address of the ServiceControl instance</param>
         /// <param name="interval">Interval between consecutive reports</param>
+        /// <param name="instanceId">Unique identifier of endpoint physical instance</param>
         [ObsoleteEx(Message = "Not for public use.")]
-        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval)
+        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval, string instanceId)
```

Comment:
> This is not good enough to show on the UI as for endpoints running on the same machine the machine name will be the same

I don't get this remark? Would this be

- multiple instances of a single logical endpoint, running on the same machine?
- multiple logical instances running on a single machine?

It seems to me [host identifier from core](https://docs.particular.net/nservicebus/hosting/override-hostid#host-identifier) would be enough? It's unlikely people will be running multiple instances of a logical endpoint on the same machine. But even then, they'd be able to create distinct names?